### PR TITLE
S171 gzipped files sanitization

### PR DESCRIPTION
**Pull request by jlorper**
### Fixes
- [proxy ndjson gzipped rules not working](https://app.asana.com/0/1195674276332239/1206890817965516)
- decompress parameter not being populated, so file handled as not compressed.

### Change implications

 - dependencies added/changed? **no**
 - something important to note in future release notes?
   - NOTE in `CHANGELOG.md` anything that will show up in `terraform plan`/`apply` that isn't
     obviously a no-op? no
   - breaking changes? if in module/example that is NOT marked `alpha`, requires major version
     change

### CODE REVIEW

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java [hunk 1]
```diff
@@ -140,12 +140,16 @@ public StorageEventRequest buildRequest(String sourceBucketName,
...
+        boolean compressedInput = compressedSource(sourceObjectPath, sourceContentEncoding);
 
         StorageEventRequest request = StorageEventRequest.builder()
             .sourceBucketName(sourceBucketName)
             .sourceObjectPath(sourceObjectPath)
             .destinationBucketName(transform.getDestinationBucketName())
             .destinationObjectPath(transform.getPathWithinBucket() + sourceObjectPathWithinBase)
+            .decompressInput(compressedInput)
+            //TODO: specify compressOutput as part of the rule. For now, follow the input
+            .compressOutput(compressedInput)
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:152]: Missing parameters, so treated input as uncompressed always

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java [hunk 2]
```diff
@@ -307,18 +310,16 @@ void validate(StorageEventRequest request,
...
 
-            StringBuffer firstLines = new StringBuffer();
-            for (int lineCount = 0; lineCount < LINES_TO_VALIDATE ; lineCount++) {
-                String line = reader.readLine();
-                if (line == null) {
-                    break;
-                }
-                firstLines.append(line + "\n");
-            }
+            String firstLines = reader.lines()
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:313]: Some modernization of code, plus better handling of empty lines / nulls

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java [hunk 3]
```diff
@@ -307,18 +310,16 @@ void validate(StorageEventRequest request,
...
+                .filter(Objects::nonNull)
+                .limit(LINES_TO_VALIDATE)
+                .collect(Collectors.joining("\n"));
 
-            Supplier<InputStream> firstLinesSupplier = () -> new ByteArrayInputStream(firstLines.toString().getBytes());
+            Supplier<InputStream> firstLinesSupplier = () -> new ByteArrayInputStream(firstLines.getBytes());
 
-            this.process(request, transform, firstLinesSupplier, () -> out);
+            // content is already decompressed, so don't try to decompress again
+            this.process(request.withDecompressInput(false), transform, firstLinesSupplier, () -> out);
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:322]: this was failing too, for validation content is always uncompressed

File: java/core/src/main/java/co/worklytics/psoxy/storage/impl/RecordBulkDataSanitizerImpl.java [hunk 4]
```diff
@@ -68,7 +71,7 @@ public void sanitize(@NonNull Reader reader,
 
         try (BufferedReader bufferedReader = new BufferedReader(reader)) {
             String line;
-            while ((line = bufferedReader.readLine()) != null) {
+            while ((line = StringUtils.trimToNull(bufferedReader.readLine())) != null) {
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/storage/impl/RecordBulkDataSanitizerImpl.java:74]: Deal with empty lines

File: java/core/src/test/java/co/worklytics/psoxy/storage/impl/RecordBulkDataSanitizerImplTest.java [hunk 5]
```diff
@@ -142,4 +143,34 @@ void noTransforms() throws IOException {
         assertEquals(EXPECTED, output);
     }
 
+    @Test
+    void gzippedContent() throws IOException {
```
Comment (jlorper) [inline java/core/src/test/java/co/worklytics/psoxy/storage/impl/RecordBulkDataSanitizerImplTest.java:147]: test with gzipped content

**Review by jlorper — COMMENTED** [review]
Review comments

**Review by aperez-worklytics — APPROVED** [review]

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java [hunk 6]
```diff
@@ -140,12 +140,16 @@ public StorageEventRequest buildRequest(String sourceBucketName,
                 .map(inputBasePath -> sourceObjectPath.replace(inputBasePath, ""))
                 .orElse(sourceObjectPath);
 
+        boolean compressedInput = compressedSource(sourceObjectPath, sourceContentEncoding);
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:143]: imho, clearer to call it `isInputCompressed`

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java [hunk 7]
```diff
@@ -363,12 +364,15 @@ void process(StorageEventRequest request,
 
     }
 
+    boolean compressedSource(String fileName, String contentEncoding) {
+        return StringUtils.equals(contentEncoding, CONTENT_ENCODING_GZIP) || fileName.endsWith(".gz");
```
Comment (eschultink) [inline java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:368]: should match canonical/standards compliant behavior - relying ONLY on content encoding; not using file extension to control the behavior (our code will warn if `.gz` ending w non-gzip encoding)

File: java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java (hunk 1 above)
Comment (eschultink) [reply java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java:152]: i prefer `inputIsCompressed` or `isCompressedInput`. `compressedInput` sounds like it's a compressed stream of the input or something.

**Pull request merged**